yt_dlp/extractor/ximalaya.py

```diff
@@ -82,6 +114,24 @@ def _real_extract(self, url):
             audio_info_file, audio_id,
             f'Downloading info json {audio_info_file}', 'Unable to download info file')
 
+        # NOTE(xcsong): VIP-restricted audio
+        if audio_info.get('is_paid'):
+            ts = int(time.time())
+            audio_info_file = f'{scheme}://mpay.ximalaya.com/mobile/track/pay/{audio_id}/{ts}?device=pc&isBackend=true&_={ts}'
+            audio_info = self._download_json(
+                audio_info_file, audio_id,
+                f'Downloading VIP info json {audio_info_file}', 'Unable to download VIP info file')
```

Comment:
Same thing here:
```suggestion
            audio_info = self._download_json(
                f'{scheme}://mpay.ximalaya.com/mobile/track/pay/{audio_id}/{ts}',
                audio_id, 'Downloading VIP info json', 'Unable to download VIP info file',
                query={'device': 'pc', 'isBackend': 'true', '_': ts})
```
Question: Do we really want to overwrite `audio_info`? This will overwrite all data from the original call. Should we extract both separately, or are all formats/metadata of the free API also included in the paid api? Should we still continue with extraction using the free api data if the users is not authenticated? i.e. are are sometimes only the higher quality formats locked behind a paid account?

Reply:
here are some response example querying a paid audio using either free API or paid API:

- free API https://m.ximalaya.com/tracks/749837049.json

![image](https://github.com/user-attachments/assets/d0047bb1-7c31-4db5-8962-67ded8f615f7)


- paid API https://mpay.ximalaya.com/mobile/track/pay/749837049/1725762545?device=pc&isBackend=true&_=1725762545

![image](https://github.com/user-attachments/assets/cd572307-b692-44ee-996e-e11e5768411d)



1. as you can see, querying a paid audio using free API result in null `play_path_64` and `play_path_32`, both higher and lower quality formats are locked, we need paid API to extract the play path.
2. you are right, they are not equal, not all metadata of the free API are included in paid API, I agree that we can extract them separately, or simply incorporate `audio_vip_info` as a metadata field within `audio_info` ?